resources/packs/azure/azure.go

```diff
@@ -16,8 +16,17 @@ func init() {
 	Registry.Add(core.Registry)
 }
 
-func azuretransport(t providers.Instance) (*azure_transport.Provider, error) {
-	at, ok := t.(*azure_transport.Provider)
+func azureTransport(t providers.Instance) (*microsoft_transport.Provider, error) {
+	at, ok := t.(*microsoft_transport.Provider)
+	if !ok || len(at.SubscriptionID()) == 0 {
+		return nil, errors.New("azure resource is not supported on this transport")
+	}
+	return at, nil
+}
+
+// temporary second function to be used only in azuread.* resources. for these, a subscription is not required.
+func msGraphTransport(t providers.Instance) (*microsoft_transport.Provider, error) {
```

Comment:
Note: This also works with the `azureTransport`, we just get a different error then (sub id cannot be empty). I figured we migh want a more explicit error if you do `azurerm.vms` from an ms365 shell

Reply:
Lets clearly mark it as `//TODO:` so that the editor hightlights it